packages/renderer/src/lib/kube/resource-listen.ts

```diff
@@ -0,0 +1,132 @@
+  // called every time events attached to the resource are updated
+  onEventsUpdated?: (events: CoreV1Event[]) => void;
+}
+
+/** listenResource listens for a specific resource (resourceName, name, namespace)
+ *  and fires events when:
+ *  - the resource is not found
+ *  - the resource is updated (will send the resource without managedFields)
+ *  - the events attached to the resource are updated, optionally
+ */
+export async function listenResource(options: ListenResourceOptions): Promise<IDisposable | undefined> {
+  let resourcesListener: IDisposable | undefined;
+  let eventsListener: IDisposable | undefined;
+
+  let resourcesUnsubscriber: Unsubscriber;
+  let eventsUnsubscriber: Unsubscriber;
+
+  let allEvents: KubernetesObject[] = [];
+  let resource: KubernetesObject | undefined;
+
+  const refreshEvents = (): void => {
+    if (options.listenEvents) {
+      const events = allEvents
+        .map(ev => (isCoreV1Event(ev) ? ev : undefined))
+        .filter(ev => !!ev)
+        .filter(ev => ev.involvedObject.uid === resource?.metadata?.uid);
+      options.onEventsUpdated?.(events);
+    }
+  };
+
+  const onUpdatedResources = (updatedResources: KubernetesObject[]): void => {
+    const matchingResource = updatedResources.find(
+      srv =>
+        srv.metadata?.name === options.name && (!options.namespace || srv.metadata?.namespace === options.namespace),
+    );
+    if (matchingResource) {
+      resource = matchingResource;
+      const kubeResource = nonVerbose(matchingResource);
+      options.onResourceUpdated(kubeResource, isExperimental);
+      refreshEvents();
+    } else {
+      options.onResourceNotFound();
+    }
+  };
+
+  const isExperimental = await isKubernetesExperimentalMode();
+  if (isExperimental) {
+    resourcesListener = await listenResources(options.resourceName, {}, onUpdatedResources);
+    if (options.listenEvents) {
+      eventsListener = await listenResources('events', {}, (updatedEvents: KubernetesObject[]) => {
+        allEvents = updatedEvents;
+        refreshEvents();
+      });
+    }
+  } else {
+    if (options.legacyResourceStore) {
+      resourcesUnsubscriber = options.legacyResourceStore.subscribe(onUpdatedResources);
+    }
+    if (options.listenEvents && options.legacyEventsStore) {
+      eventsUnsubscriber = options.legacyEventsStore.subscribe(updatedEvents => {
+        allEvents = updatedEvents.map(e => ({ kind: 'Event', ...e }));
+        refreshEvents();
+      });
+    }
+  }
+
+  return {
+    dispose: (): void => {
+      resourcesListener?.dispose();
+      eventsListener?.dispose();
+      resourcesUnsubscriber?.();
+      eventsUnsubscriber?.();
+    },
+  };
+}
+
+// removes "verbose" fields from a Kubernetes resource description:
+// - metadata.managedFields
+function nonVerbose(resource: KubernetesObject): KubernetesObject {
+  if (resource.metadata?.managedFields) {
+    const result = structuredClone(resource);
```

Comment:
Why using `structuredClone`, can't we just mutate the original object ?

Reply:
The caller may want to keep the verbose object

Reply:
I've made some back and forth on this: I'm still wondering if `listenResource` should return the non verbose version, or if instead `nonVerbose` should be exported and called by the Details component, to send the non verbose version to the YAML Editor, and keep the verbose for the Summary/Inspect components.